### DISCUSSION TITLE
Integration of Gemini Model in Open Web UI: Issues with RAG and Citation Functionality

### DISCUSSION THREAD

**Discussion by sjoonb** [open-webui/open-webui#3660] [General]
I want to apply the Gemini model to Open Web UI, but it doesn’t work well. When I checked issue #3482, I tried using Functions and Pipelines, which resulted in generating responses, but the existing RAG functionality and Citation functionality do not work smoothly.

**Comment by ds0nt** [comment]
Is it possible to put Gemini in the connections dropdown? That seems like it would be suitable.

Maybe I can take a swing at it? Is it feasible?

**Reply by ds0nt** [reply]
It looks cool
![2024-08-12-142421_1436x1184_scrot](https://github.com/user-attachments/assets/0cc55f76-2a3c-474b-b408-31bc9d9c9c03)

**Comment by justinh-rahb** [comment]
The Functions and Pipelines variant of this manifold have both been pretty solid.

https://openwebui.com/f/justinrahb/google_genai

https://raw.githubusercontent.com/open-webui/pipelines/main/examples/pipelines/providers/google_manifold_pipeline.py

**Comment by samyogdhital** [comment]
@justinh-rahb can we make text streaming possible in this function?

**Comment by YuriyGavrilov** [comment]
+1 to gemini, wish to add gemini like openai with custom url as proxy.
